src/data/real/basic.lean

```diff
@@ -103,26 +103,34 @@ lemma cauchy_nat_cast (n : ℕ) : (n : ℝ).cauchy = n := rfl
 lemma cauchy_int_cast (z : ℤ) : (z : ℝ).cauchy = z := rfl
 lemma cauchy_rat_cast (q : ℚ) : (q : ℝ).cauchy = q := rfl
 
+instance has_smul {α} [has_smul α ℚ] [is_scalar_tower α ℚ ℚ] : has_smul α ℝ :=
+{ smul := λ a r, ⟨a • r.cauchy⟩ }
+lemma of_cauchy_smul {α} [has_smul α ℚ] [is_scalar_tower α ℚ ℚ] (a : α) (q) :
+  (⟨a • q⟩ : ℝ) = a • ⟨q⟩ := rfl
+lemma cauchy_smul {α} [has_smul α ℚ] [is_scalar_tower α ℚ ℚ] (a : α) (r : ℝ) :
+  (a • r).cauchy = a • r.cauchy := rfl
+
+/--
+This instance has low priority so that `norm_num` finds `monoid.has_pow` instead of it.
+The two are defeq, but `norm_num` matches up to syntactic equality.
+-/
+@[priority 50]
+instance has_nat_pow : has_pow ℝ ℕ := { pow := λ r n, ⟨r.cauchy ^ n⟩ }
```

Comment:
Why should this be an instance at all, if it's causing this trouble?

Reply:
It's used by the call to `function.surjective.comm_ring` below. The instsance is perfectly safe, there's just a bug in `.norm_num`.

I could switch to `local attribute [instance]` if you prefer?

Reply:
I think I would prefer the `local attribute` path.

Reply:
I've pushed that at https://github.com/leanprover-community/mathlib/tree/eric-wieser/real.comm_ring-alt so that I can compare the CI output in case only one approach fails.

Reply:
Alternatively, I could just fix the norm_num bug: #18129